packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -850,6 +864,27 @@ abstract class CustomArrayNodeBase<const T extends ImplicitAllowedTypes>
 			destinationIndex,
 		);
 	}
+
+	public values(): IterableIterator<TreeNodeFromImplicitAllowedTypes<T>> {
+		return this.generateValues(this.#generationNumber);
+	}
+	private *generateValues(
+		initialLastUpdatedStamp: number,
+	): Generator<TreeNodeFromImplicitAllowedTypes<T>> {
+		if (initialLastUpdatedStamp < this.#generationNumber) {
+			throw new UsageError(`Concurrent editing and iteration is not allowed.`);
+		}
+		for (
+			let i = 0, value = this.at(0);
+			i < this.length && value !== undefined;
```

Comment:
```suggestion
			i < this.length;
```
`value` will always be defined since we don't support sparse arrays, right?

Reply:
I did remove the `&& value !== undefined`, but typescript seems to be complaining in the for loop that we cannot yield undefined values (we can only return `TreeNodeFromImplicitAllowedTypes<T>` because of our return type).

I updated to use null assertions, but wasn't sure if this was the best way to do it. Would it be better if I added an assert statement in the for loop instead?

Reply:
Discussed offline.